### PR TITLE
JSAB2-25 Registration frontend

### PR DESCRIPTION
**Pull request by HanqingZ**
- add registration page with a simple verification for email and a blacklist for username
- PLEASE use the branch JSAB2-25-registration-frontend-test for testing

Scenario 1:
Given the /register route.
When the page is loaded.
Then the register form exists.

Scenario 2:
Given the loaded register page.
When I click on the submit button.
And the input fields are blank.
Then an error message shows up: "Username and password are required."

Scenario 4:
Given the loaded register page.
When I click on the submit button.
And the password is under 8 characters.
Then an error message shows up: "Password must be at least 8 characters."

Scenario 5:
Given the loaded register page.
When I click on the submit button.
And I give a unique user name
And I give an 8 character long password
And I optionally give a kingdom name
Then I am navigated to the /maps page. (mocked if not existing yet)

### CODE REVIEW

**Review by GuJialu — CHANGES_REQUESTED** [review]
Still need password length check and mocked map page.

**Review by GuJialu — APPROVED** [review]

**Review by GuJialu — APPROVED** [review]

**Comment by kyya** [comment]
Something has been checked in JSAB2-90, Login page addition.

**Review by kyya — APPROVED** [review]

**Review by GuJialu — APPROVED** [review]

**Review by kyya — APPROVED** [review]

File: frontend/src/Registration/index.jsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    }
+    dispatch(registrationSuccess());
+    if (registerError !== undefined) {
+      return showAlert(`Oops, ${registerError.message}`);
+    }
+    if (isLoading) {
+      return <ActivityIndicator size="large" color={Colors.tealColor} />;
+    }
+    showAlert('Registration Success.');
+    return navigation.navigate('Map');
```
Comment (evasimonyi) [inline frontend/src/Registration/index.jsx:83]: this part doesn't work for me, neither on Android or on iPhone, I just get a warning about the successful registration, but it doesn't redirect me to the mocked Map page, is it working on a simulator or tablet? can you please check it?

**Review by evasimonyi — CHANGES_REQUESTED** [review]

**Review by HONGTAO-T — APPROVED** [review]

**Review by kyya — APPROVED** [review]

**Review by evasimonyi — APPROVED** [review]
👍 💯

**Pull request merged**